src/Entity/PaymentStateData/PaymentStateDataEntity.php

```diff
@@ -0,0 +1,96 @@
+<?php declare(strict_types=1);
+/**
+ *                       ######
+ *                       ######
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen Payment Module
+ *
+ * Copyright (c) 2020 Adyen B.V.
+ * This file is open source and available under the MIT license.
+ * See the LICENSE file for more info.
+ *
+ * Author: Adyen <shopware@adyen.com>
+ */
+
+namespace Adyen\Shopware\Entity\PaymentStateData;
+
+use Shopware\Core\Framework\DataAbstractionLayer\Entity;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityIdTrait;
+
+class PaymentStateDataEntity extends Entity
+{
+    use EntityIdTrait;
+
+    /**
+     * @var string
+     */
+    protected $token;
+
+    /**
+     * @var string
+     */
+    protected $statedata;
+
+    /**
+     * @var string
+     */
+    protected $origin;
+
+    /**
+     * @return string
+     */
+    public function getToken(): string
+    {
+        return $this->token;
+    }
+
+    /**
+     * @param string $token
+     */
+    public function setToken(string $token): void
+    {
+        $this->token = $token;
+    }
+
+    /**
+     * @return string
+     */
+    public function getStateData(): string
+    {
+        return $this->statedata;
+    }
+
+    /**
+     * @param string $stateData
+     */
+    public function setStateData(string $statedata): void
+    {
+        $this->statedata = $statedata;
+    }
+
+    /**
+     * @return string
+     */
+    public function getOrigin(): string
```

Comment:
wouldn't it be better to have an additionalData field where we could store the extra details? It can be origin or type or guestEmail or anything that is not sent along with the state data but still necessary - I don't expect to have too many of these but if we have another one it's already extra work to add it to the db. Let me know what do you think.

Reply:
Removed the origin column and is now included in state.data additionalData